src/extensions/surveys.ts

```diff
@@ -0,0 +1,556 @@
+                    if (diffDaysFromToday < surveyWaitPeriodInDays) {
+                        return
+                    }
+                }
+
+                if (!localStorage.getItem(`seenSurvey_${survey.id}`)) {
+                    let surveyPopup
+                    const surveyQuestionType = survey.questions[0].type
+                    if (surveyQuestionType === 'rating') {
+                        surveyPopup = createRatingsPopup(posthog, survey)
+                    } else if (surveyQuestionType === 'open' || surveyQuestionType === 'link') {
+                        surveyPopup = createOpenTextPopup(posthog, survey)
+                    } else if (surveyQuestionType === 'single_choice' || surveyQuestionType === 'multiple_choice') {
+                        surveyPopup = createMultipleChoicePopup(posthog, survey)
+                    }
+
+                    if (!surveyPopup) {
+                        console.error(`PostHog: Survey question type: ${surveyQuestionType} not supported`)
+                        return
+                    }
+
+                    const shadow = createShadow(style(survey.id, survey?.appearance), survey.id)
+
+                    addCancelListeners(posthog, surveyPopup, survey.id, survey.name)
+                    if (survey.appearance?.whiteLabel) {
+                        ;(
+                            surveyPopup.getElementsByClassName('footer-branding') as HTMLCollectionOf<HTMLElement>
+                        )[0].style.display = 'none'
+                    }
+                    shadow.appendChild(surveyPopup)
+
+                    window.dispatchEvent(new Event('PHSurveyShown'))
+                    posthog.capture('survey shown', {
+                        $survey_name: survey.name,
+                        $survey_id: survey.id,
+                        sessionRecordingUrl: posthog.get_session_replay_url(),
+                    })
+                    localStorage.setItem(`lastSeenSurveyDate`, new Date().toISOString())
+                    if (survey.appearance?.displayThankYouMessage) {
+                        window.addEventListener('PHSurveySent', () => {
+                            const thankYouElement = createThankYouMessage(survey)
+                            shadow.appendChild(thankYouElement)
+                            window.setTimeout(() => {
+                                thankYouElement.remove()
+                            }, 2000)
+                        })
+                    }
+                }
+            }
+        })
+    }, forceReload)
+}
+
+export function generateSurveys(posthog: PostHog) {
+    callSurveys(posthog, true)
+
+    let currentUrl = location.href
+    if (location.href) {
+        setInterval(() => {
+            if (location.href !== currentUrl) {
+                currentUrl = location.href
+                callSurveys(posthog, false)
+            }
+        }, 1500)
```

Comment:
Opinion incoming - this feels like something that should be in `posthog-js` - like `posthog.onActiveSurveysChanged()` or something. Or is it somehow unique to our GUI implementation?

Reply:
Makes some sense, will skip for now though and have a think after the migration